### ISSUE TITLE
Update game player panel.

### ISSUE THREAD

**Issue by lodici** [magarena/magarena#102]
![screenshot00065](https://cloud.githubusercontent.com/assets/2443519/6641088/8f7f6d40-c993-11e4-8d4d-fcba50cd81b8.png)
- [x] display Persistent health overlay at bottom left of player image.
- [x] display Poison counter at bottom right of player image if poison damage > 0.
- [x] display Damage prevention counter at top right of player image if damage prevention > 0.
- [x] display player name and AI level and type.
- [x] re-style player zone counters as toggle buttons
- [x] highlight active zone button.
- [x] click zone toggle button to display associated zone cards in bottom area of game board.
- [x] click already selected zone toggle button to display full screen zone view.
- [x] display zone name on mouse over toggle button.
- [x] ~~also display deck name on mouse over library button.~~
- [x] set player image to greyscale when not player's turn.
- [x] display active zone name on board unless it is the user's hand.
- [x] "S" switch key switches zones instead of players.

**Comment by ShawnieBoy** [comment]
I'm liking this a lot - :smiley:

**Comment by lodici** [comment]
It is all sitting on a feature branch so I hope to merge into master over the next few days (GitHub's task markdown syntax is pretty cool!). If there is anything you feel is missing please let me know while I am focused on the side-bar area of the game board screen.

**Comment by ShawnieBoy** [comment]
Cool - I don't know if this is already in the pipeline, but in the log, the referenced card names in brackets (from searching etc) Having them like a hyperlink-card-pop-up, for when the AI 'reveals' a card you're not quite sure of.

**Comment by melvinzhang** [comment]
The poison and damage prevention isn't obvious at first glance. I think we
have some icons for them on creatures. Perhaps we can reuse the icons for
the player.

**Comment by lodici** [comment]
> Cool - I don't know if this is already in the pipeline, but in the log, the referenced card names in brackets (from searching etc) Having them like a hyperlink-card-pop-up, for when the AI 'reveals' a card you're not quite sure of.

This probably deserves its own issue. I am fairly sure there is an issue from  the previous tracker along the same lines. I think this would be a very useful feature and would be consistent with the way the stack and user action panel work. I also think the reason it has not been implemented yet is that the log has quite a bit of tricksy code for layout and formatting each log entry - one of the reasons why there is still no scrollbar.

> The poison and damage prevention isn't obvious at first glance. I think we have some icons for them on creatures. Perhaps we can reuse the icons for the player.

I went for simple as a first attempt but there is no reason why these could not be icons. I did wonder if the poison counter should not be red but for some reason I associate green (although perhaps a bit more sickly than shown). Ultimately, the counter style and zone icons should be probably be integrated into the theme settings. Can't guarantee for 1.60 though.

**Comment by lodici** [comment]
Display active zone name on game board unless it is the user's hand.
![screenshot00055](https://cloud.githubusercontent.com/assets/2443519/6694201/7935d228-ccd0-11e4-95e7-c85f7e408a40.png)

**Comment by lodici** [comment]
Zone toggle button associated with active zone is highlighted and animated on select.
![zonebuttons](https://cloud.githubusercontent.com/assets/2443519/6720297/9e888c3e-cdb9-11e4-8c52-3c281faf3ad7.gif)

**Comment by ShawnieBoy** [comment]
This looks really good! - I'm guessing that the tiny button-bar could be entirely optional if the larger images function as buttons. (Still keeping that great animation)

:+1:

**Comment by lodici** [comment]
> I'm guessing that the tiny button-bar could be entirely optional if the larger images function as buttons.

You wouldn't believe the hoops I am having to jump through to fit the around the damn button bar! :stuck_out_tongue_closed_eyes: The thing is, it is actually used in TextMode (horizontally) as well so I have to tread lightly. I do hope eventually we can send it (and the nebulous ``other`` button) the way of the dodo and use the zone buttons to replicate.

**Comment by lodici** [comment]
Now that we have a clickable "Hand" zone button for each player I have changed the behaviour of the switch ("S") key to switch the selected zone between players instead of swapping the actual players - I think this is much more consistent (and much less of a hack code-wise than the previous option).

**Comment by lodici** [comment]
Decided to leave the deck name tooltip for the time being - bit of a pain.

**Comment by lodici** [comment]
I have decided to use a simple outline to indicate the active zone.
![screenshot00083](https://cloud.githubusercontent.com/assets/2443519/6881395/7f3f734e-d555-11e4-9fe1-b397a44fca08.png)
Because it is persistent I feel the yellow fill and border is too strident. I have also removed the animation because I would prefer to use button FX sparingly so as to clearly alert the user. For example, if a zone is programatically selected then this would be good point to flash the icon so as to draw the user's attention. Another example, is if you have the choice to select cards from multiple zones then we can use another FX to indicate which zones have valid cards. Having a simple "normal" state makes it easier to add FX on top.
